backend/tracim_backend/lib/core/user.py

```diff
@@ -226,25 +226,35 @@ def get_known_user(
         # should show only user in same workspace as user
         assert not (filter_results and not self._user)
         if filter_results and self._user and self._user.profile.id <= Profile.USER.id:
-            user_workspaces_id_query = (
-                self._session.query(UserRoleInWorkspace.workspace_id)
-                .distinct(UserRoleInWorkspace.workspace_id)
-                .filter(UserRoleInWorkspace.user_id == self._user.user_id)
-            )
-            users_in_workspaces = (
-                self._session.query(UserRoleInWorkspace.user_id)
-                .distinct(UserRoleInWorkspace.user_id)
-                .filter(UserRoleInWorkspace.workspace_id.in_(user_workspaces_id_query.subquery()))
-                .subquery()
-            )
+            users_in_workspaces = self._get_user_ids_in_same_workspace(self._user.user_id)
             query = query.filter(User.user_id.in_(users_in_workspaces))
         if exclude_user_ids:
             query = query.filter(~User.user_id.in_(exclude_user_ids))
         query = query.limit(nb_elem)
         return query.all()
 
-    # Check methods
+    def _get_user_ids_in_same_workspace(self, user_id: int):
+        user_workspaces_id_query = (
+            self._session.query(UserRoleInWorkspace.workspace_id)
+            .distinct(UserRoleInWorkspace.workspace_id)
```

Comment:
Is `distinct` necessary here ? (which would imply that a user can have several roles in a workspace).

Reply:
This code is not really new, i just moved it in order to reuse it, so i didn't check this.

UserRoleInWorkspace Table contain both user_id and workspace id as primary key, so we can have multiple item with different workspace id. It's right that normally, if we do select for user, there is no reason to get multiple item for same workspace id (it will brake primary key unicity).

Reply:
removed.